### PR TITLE
[query] Uncomment fixed promql test cases

### PR DESCRIPTION
**Pull request by linasm**
**What this PR does / why we need it**:
This enables some of the promql tests that were previously commented out.

**Special notes for your reviewer**:
Had to do some additional changes in `m3comparator` to support some of those test queries.

**Does this PR introduce a user-facing and/or backwards incompatible change?**:
NONE

**Does this PR require updating code package or user-facing documentation?**:
NONE

### CODE REVIEW

File: src/query/test/testdata/literals.test [hunk 1]
```diff
@@ -1,61 +0,0 @@
-# FAILING issue #46 (we return vector resultType in place of scalar).
```
Comment (robskillington) [inline src/query/test/testdata/literals.test:1]: Hm why are we removing the literals test?

Comment (robskillington) [reply src/query/test/testdata/literals.test:1]: Oh it got moved 👍

File: src/query/test/promql/m3comparator_client.go [hunk 2]
```diff
@@ -18,7 +18,7 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package test
+package promql
```
Comment (arnikola) [inline src/query/test/promql/m3comparator_client.go:21]: Hmm not too sure about the package name here, it's not really promql but more comparator stuff

Comment (gediminasgu) [reply src/query/test/promql/m3comparator_client.go:21]: `compatibility`tests package?

Comment (linasm) [reply src/query/test/promql/m3comparator_client.go]: Yeah, `compatibility` sounds just about right.

File: src/query/test/compatibility/test.go [hunk 3]
```diff
@@ -560,7 +562,7 @@ func almostEqual(a, b float64) bool {
 	diff := math.Abs(a - b)
 
 	if a == 0 || b == 0 || diff < minNormal {
-		return diff < epsilon*minNormal
+		return diff < epsilon
```
Comment (arnikola) [inline src/query/test/compatibility/test.go:565]: nit: can delete `minNormal` completely I think

Comment (linasm) [reply src/query/test/compatibility/test.go:565]: Not sure about this. It seems that this is designed to catch near-zero values in these tests:
https://github.com/m3db/m3/blob/master/src/query/test/testdata/aggregators.test#L97

Comment (arnikola) [reply src/query/test/compatibility/test.go:565]: Ah yeah missed that it was used on :564

**Pull request merged**